client/src/main/scala/drt/client/services/handlers/FixedPointsHandler.scala

```diff
@@ -15,45 +16,55 @@ import scala.concurrent.duration._
 import scala.language.postfixOps
 import scala.scalajs.concurrent.JSExecutionContext.Implicits.queue
 
-class FixedPointsHandler[M](viewMode: () => ViewMode, modelRW: ModelRW[M, Pot[FixedPointAssignments]]) extends LoggingActionHandler(modelRW) {
+class FixedPointsHandler[M](getCurrentViewMode: () => ViewMode, modelRW: ModelRW[M, Pot[FixedPointAssignments]]) extends LoggingActionHandler(modelRW) {
+
+  def startMillisFromView: MillisSinceEpoch = getCurrentViewMode().dayStart.millisSinceEpoch
+
+  def viewHasChanged(viewMode: ViewMode): Boolean = viewMode.dayStart.millisSinceEpoch != startMillisFromView
+
+  def scheduledRequest(viewMode: ViewMode): Effect = Effect(Future(GetFixedPoints(viewMode))).after(2 seconds)
+
   protected def handle: PartialFunction[Any, ActionResult[M]] = {
-    case SetFixedPoints(fixedPoints, _) => updated(Ready(fixedPoints))
+    case SetFixedPoints(viewMode, fixedPoints, _) =>
+      if (viewMode.isHistoric)
+        updated(Ready(fixedPoints))
+      else
+        updated(Ready(fixedPoints), scheduledRequest(viewMode))
 
     case SaveFixedPoints(assignments, terminalName) =>
       log.info(s"Calling saveFixedPoints")
 
       val otherTerminalFixedPoints = value.getOrElse(FixedPointAssignments.empty).notForTerminal(terminalName)
       val newFixedPoints: FixedPointAssignments = assignments + otherTerminalFixedPoints
       val futureResponse = DrtApi.post("fixed-points", write(newFixedPoints))
-        .map(_ => SetFixedPoints(newFixedPoints, Option(terminalName)))
+        .map(_ => NoAction)
         .recoverWith {
           case _ =>
             log.error(s"Failed to save FixedPoints. Re-requesting after ${PollDelay.recoveryDelay}")
             Future(RetryActionAfter(SaveFixedPoints(assignments, terminalName), PollDelay.recoveryDelay))
         }
       effectOnly(Effect(futureResponse))
 
-    case GetFixedPoints() =>
-      val fixedPointsEffect = Effect(Future(GetFixedPoints())).after(60 minutes)
-      log.info(s"Calling getFixedPoints")
+    case GetFixedPoints(viewMode) if viewHasChanged(viewMode) =>
+      log.info(s"Ignoring old view response")
+      noChange
+
+    case GetFixedPoints(viewMode) =>
+      val maybePointInTimeMillis = if (viewMode.isHistoric) Option(viewMode.millis) else None
```

Comment:
this `if` and the below `match` can probably be rolled into one.

Reply:
Yup